src/app/esp8266/tcpnew.fth

```diff
@@ -75,6 +75,7 @@ defer respond   ( pcb -- close? )
    ?dup 0=  if                  ( pbuf )
       ." Connection closed" cr cr
       rx-pcb close-connection   ( )
+      ERR_ABRT
```

Comment:
I think that ERR_OK is appropriate here.  This is not a premature abort, but rather a normal shutdown.

That said, looking at the code in lwip/core/tcp_in.c, there is very little difference between the code paths in the two cases.  The only difference is that, in the OK case, tcp_output() is called before falling through to the aborted: label.  I think that is a good thing, as it permits the stack to proceed apace with the final stages of the connection close dance.

Regarding the timing of tcp_recved versus handle-data, it's unclear to me what is correct.  On the one hand, calling tcp_recved early would potentially speed things up by getting the TCP ACK out sooner.  On the other hand, it seems prudent to make sure the data is safe before proceeding.  On the other other hand, the data is probably safe until the pbuf is freed.

Analysis of tcp_in.c suggests that, before the pbuf is passed to the application, it is disconnected from the pcb, so it should be safe to call tcp_recved before handling the data.  That being the case, I have moved the tcp-recved call up to before handle-data.  We can keep our fingers crossed.